### DISCUSSION TITLE
Library not detecting face correctly

### DISCUSSION THREAD

**Discussion by rael346** [ChenglongMa/SkinToneClassifier#13] [Q&A] [answered]
Hi, 

So I have been using the tool to classify skin tones for images that were generated by Stable Diffusion. For some of the images the auto color detection (whether the image is black & white or color) is not correct, since they detect the wrong facial area (see the below image and debugs for example).

<img src="https://github.com/ChenglongMa/SkinToneClassifier/assets/75349763/71e32d32-5173-4bbd-bbed-15a52d9ebc0e" width="250" height="250"/>

<img src="https://github.com/ChenglongMa/SkinToneClassifier/assets/75349763/d8620730-f925-4e7d-ac9e-e67234c392ff" width="350" height="250"/>

<img src="https://github.com/ChenglongMa/SkinToneClassifier/assets/75349763/6f28a449-c4bc-4277-9469-187e3d8f2fbc" width="350" height="250"/>

Is there a way that we can know when the image doesn't detect the facial area correctly (it doesn't form a square around the face but instead the rough areas like in the third image)?

**Comment by rael346** [comment, marked as answer]
Nevermind, I found it in the documentation for the `--debug` flag
> This option will store the report image (like the Lenna example above) in `./path/to/output/debug/<image type>/faces_<n>` folder, where `<image type>` indicates if the image is color or bw (black/white); `<n>` is the number of faces detected in the image.

**Comment by rael346** [comment]
On a different note I saw that most of the images that I classify, most of them got classified as black & white (I have 1700 images, and only 1 of them was classified as color, even though most of the images are color). So there might be a bug for the `auto` behavior

**Reply by ChenglongMa** [reply]
Hi @rael346,

Thanks for your feedback!
I will keep improving the algorithm to correctly detect the image type. May I know what data set you're using? Or would you mind sharing a few misclassified images so that I can debug the algorithm?

Thanks!

PS: According to the result in your original question, the algorithm will square around the face if it found a "face area", otherwise, it will show what it thinks is the "area of ​​skin".

**Reply by rael346** [reply]
Hi @ChenglongMa,

Here are some of the images that the algorithm misclassified. I generated these from Stable Diffusion V2.1 on Hugging Face using the prompt "A colored photograph of a [job], headshot, high quality", with job being some job that I'm doing an audit on (software engineer, nurse, etc).

<img src="https://github.com/ChenglongMa/SkinToneClassifier/assets/75349763/7e5cb1d9-b5b3-49a9-94aa-ad26a1a9e392" width=300 height=300>

<img src="https://github.com/ChenglongMa/SkinToneClassifier/assets/75349763/93693f86-fd11-4f6d-98c5-69d45aeb6196" width=300 height=300>

<img src="https://github.com/ChenglongMa/SkinToneClassifier/assets/75349763/dd00691b-5901-4ba8-a8b6-6e9d78c7bfac" width=300 height=300>

<img src="https://github.com/ChenglongMa/SkinToneClassifier/assets/75349763/63276fce-a7df-4a87-be84-4bc39a1292ee" width=300 height=300>

**Reply by ChenglongMa** [reply]
Thanks for your images. I will debug the algorithm soon.
